### PR TITLE
Merge asm.i8080 into anal.i8080

### PR DESCRIPTION
**Pull request by terorie**
- [ ] Mark this if you consider it ready to merge
- [ ] I've added tests (optional)
- [ ] I wrote some lines in the [book](https://github.com/radareorg/radare2book) (optional)

**Description**

- Merge asm.i8080 into anal.i8080 ##refactor
- Add asm i8080 test

### CODE REVIEW

**Comment by trufae** [comment]
Missing opasm i think

ADD61527-EE7A-4D3F-8783-E0E55F25455A

**Comment by terorie** [comment]
@radare the CI failure was caused by an invalid test that attempted to disassemble two bytes for a three bytes opcode. It worked fine on my machine but not in CI due to memory corruption. Fixed now.

**Comment by trufae** [comment]
strndup is not portable. Use r_str_ndup
Creating library libr/anal\r_anal.lib and object libr/anal\r_anal.exp
[3869](https://github.com/radareorg/radare2/runs/6932386049?check_suite_focus=true#step:7:3870)
p_anal_i8080.c.obj : error LNK2019: unresolved external symbol strndup referenced in function i8080_op
[3870](https://github.com/radareorg/radare2/runs/6932386049?check_suite_focus=true#step:7:3871)
libr\anal\r_anal.dll : fatal error LNK1120: 1 unresolved externals

File: libr/anal/p/anal_i8080.c [hunk 1]
```diff
@@ -14,6 +14,9 @@
 static int i8080_op(RAnal *anal, RAnalOp *op, ut64 addr, const ut8 *data, int len, RAnalOpMask mask) {
 	char out[32];
 	int ilen = i8080_disasm (data, out, len);
+	if (mask & R_ANAL_OP_MASK_DISASM) {
+		op->mnemonic = strndup (out, sizeof(out));
```
Comment (trufae) [inline libr/anal/p/anal_i8080.c:18]:
```suggestion
		op->mnemonic = r_str_ndup (out, sizeof(out));
```

**Comment by trufae** [comment]
oopsl the assembler wasnt handled?

Screenshot 2022-06-17 at 17 17 11

**Pull request merged**